src/helpers/github.ts

```diff
@@ -99,3 +99,66 @@ export async function isCiGreen({ octokit, logger, env }: Context, sha: string,
     return false;
   }
 }
+
+function parseTarget({ payload }: Context, target: string) {
+  const owner = payload.repository.owner?.login || "";
+  const [orgParsed, repoParsed] = target.split("/");
+  let repoTarget = null;
+  if (orgParsed !== owner) {
+    return null;
+  } else if (repoParsed) {
+    repoTarget = repoParsed;
+  }
+  return { org: owner, repo: repoTarget };
+}
+
+/**
+ * Returns all the pull requests that are opened and not a draft from the list of repos / organizations.
+ *
+ * https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-only-issues-or-pull-requests
+ */
+export async function getOpenPullRequests(context: Context, targets: ReposWatchSettings) {
+  const { octokit, logger } = context;
+  const filter = [
+    ...targets.monitor.reduce<string[]>((acc, curr) => {
+      const parsedTarget = parseTarget(context, curr);
+      if (parsedTarget) {
+        return [...acc, parsedTarget.repo ? `repo:${parsedTarget.org}/${parsedTarget.repo}` : `org:${parsedTarget.org}`];
+      }
+      return acc;
+    }, []),
+    ...targets.ignore.reduce<string[]>((acc, curr) => {
+      const parsedTarget = parseTarget(context, curr);
+      if (parsedTarget) {
+        return [...acc, parsedTarget.repo ? `-repo:${parsedTarget.org}/${parsedTarget.repo}` : `-org:${parsedTarget.org}`];
+      }
+      return acc;
+    }, []),
+  ];
+  try {
+    const data = await octokit.paginate(octokit.rest.search.issuesAndPullRequests, {
+      q: `is:pr is:open draft:false ${filter.join(" ")}`,
+    });
+    return data.flat();
+  } catch (e) {
+    logger.error(`Error getting open pull-requests for targets: [${filter.join(", ")}]. ${e}`);
+    throw e;
+  }
+}
+
+export async function mergePullRequest(context: Context, { repo, owner, issue_number: pullNumber }: IssueParams) {
+  await context.octokit.pulls.merge({
```

Comment:
missing `rest` here and also don't we usually wrap all octokit calls with try-catch? I guess the try-catch where it's called would catch anything. More of a question but do we follow a pattern for this?

Reply:
I do not catch here to let it bubble up to the main call so errors can be sent with the whole stack. I can rethrow if needed but not sure it would be useful here.

Reply:
> More of a question but do we follow a pattern for this?

That answers that question basically. So we are sort of starting to follow the pattern of let everything bubble up to the entry then log and comment